data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/regex/RegexValueValidator.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.regex;
+
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+public class RegexValueValidator {
```

Comment:
Let's make this an actual JSR-380 validator so that we can use annotations directly for it. 

And please move it to `data-prepper-api`. I think we have other validators there.